packages/blade/src/components/Switch/ThumbIcon.tsx

```diff
@@ -0,0 +1,54 @@
+import isNumber from 'lodash/isNumber';
+import type { SwitchProps } from './types';
+import { AnimatedThumbIcon } from './AnimatedThumbIcon';
+import { switchColors } from './switchTokens';
+import { useTheme } from '~components/BladeProvider';
+import { getIn, makeSize, makeSpace, useBreakpoint } from '~utils';
+import size from '~tokens/global/size';
+import { Path } from '~components/Icons/_Svg';
+
+const switchIconSize = {
+  desktop: {
+    small: size[6],
+    medium: 'spacing.3',
+  },
+  mobile: {
+    small: 'spacing.3',
+    medium: size[10],
+  },
+} as const;
+
+const ThumbIcon = ({
+  isChecked,
+  isDisabled,
+  size = 'medium',
+}: Pick<SwitchProps, 'isChecked' | 'isDisabled' | 'size'>): React.ReactElement => {
+  const { theme } = useTheme();
+  const { matchedDeviceType } = useBreakpoint({ breakpoints: theme.breakpoints });
+
+  const width = switchIconSize[matchedDeviceType][size];
+  const height = switchIconSize[matchedDeviceType][size];
+  const finalWidth = isNumber(width) ? makeSize(width) : makeSpace(getIn(theme, width));
+  const finalHeight = isNumber(height) ? makeSize(height) : makeSpace(getIn(theme, height));
+  const variant = isDisabled ? 'disabled' : 'default';
+  const fillColor = getIn(theme, switchColors.thumbIcon[variant].fill);
+
+  return (
+    <AnimatedThumbIcon
+      isChecked={Boolean(isChecked)}
+      width={finalWidth}
+      height={finalHeight}
+      viewBox="0 0 11 8"
```

Comment:
Thats a very odd viewBox. Very different than our standard icons. Can you re-check this with design?

Reply:
Yeah checked it, it's a custom SVG shape.

Reply:
Add a comment

Reply:
Done.